src/services/WaveformDataUtils.js

```diff
@@ -161,15 +165,55 @@ export default class WaveformDataUtils {
    * @param {Object} peaksInstance - current peaks instance for the waveform
    */
   activateSegment(id, peaksInstance) {
-    const segment = peaksInstance.segments.getSegment(id);
+    const validatedPeaks = this.initialSegmentValidation(id, peaksInstance);
+    const segment = validatedPeaks.segments.getSegment(id);
     segment.update({
       editable: true,
       color: COLOR_PALETTE[2]
     });
     let startTime = segment.startTime;
     // Move play head to the start time of the selected segment
-    peaksInstance.player.seek(startTime);
+    validatedPeaks.player.seek(startTime);
 
+    return validatedPeaks;
+  }
+
+  /**
+   * Adjust segment's end time to depict the valid time range for it to be spanned
+   * At initial load of the editor for all segments with invalid end times,
+   * segment.endTime is set to duration by default
```

Comment:
Is this statement about this method? I don't see segment.endTime being set to duration in any of the cases.

Reply:
Yes, you are correct. This description was written for a different implementation. Thanks for catching this. I will fix it :+1: